stl/inc/memory

```diff
@@ -393,6 +770,95 @@ void uninitialized_value_construct(const _NoThrowFwdIt _First, const _NoThrowFwd
+            char* const _First_ch = reinterpret_cast<char*>(_First);
+            char* const _Last_ch  = reinterpret_cast<char*>(_Last);
+            _CSTD memset(_First_ch, 0, static_cast<size_t>(_Last_ch - _First_ch));
+            return _Last;
+        } else if constexpr (sized_sentinel_for<_Se, _It>) {
+            const _It _Final      = _RANGES next(_First, _Last - _First);
+            char* const _First_ch = reinterpret_cast<char*>(_First);
+            char* const _Final_ch = reinterpret_cast<char*>(_Final);
+            _CSTD memset(_First_ch, 0, static_cast<size_t>(_Final_ch - _First_ch));
+            return _Final;
+        } else {
+            const _It _Final      = _RANGES next(_First, _STD move(_Last));
+            char* const _First_ch = reinterpret_cast<char*>(_First);
+            char* const _Final_ch = reinterpret_cast<char*>(_Final);
+            _CSTD memset(_First_ch, 0, static_cast<size_t>(_Final_ch - _First_ch));
+            return _Final;
+        }
+    }
+
+    // VARIABLE ranges::uninitialized_value_construct
+    class _Uninitialized_value_construct_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <_No_throw_forward_iterator _It, _No_throw_sentinel_for<_It> _Se>
+            requires default_initializable<iter_value_t<_It>>
+        _It operator()(_It _First, _Se _Last) const {
+            // clang-format on
+            _Adl_verify_range(_First, _Last);
+            auto _UResult = _Uninitialized_value_construct_unchecked(
+                _Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)));
+
+            _Seek_wrapped(_First, _STD move(_UResult));
+            return _First;
+        }
+
+        // clang-format off
+        template <_No_throw_forward_range _Rng>
+            requires default_initializable<range_value_t<_Rng>>
+        borrowed_iterator_t<_Rng> operator()(_Rng&& _Range) const {
+            // clang-format on
+            auto _UResult = _Uninitialized_value_construct_unchecked(_Ubegin(_Range), _Uend(_Range));
+
+            return _Rewrap_iterator(_Range, _STD move(_UResult));
+        }
+
+    private:
+        template <class _It, class _Se>
+        _NODISCARD static _It _Uninitialized_value_construct_unchecked(_It _OFirst, const _Se _OLast) {
+            _STL_INTERNAL_STATIC_ASSERT(_No_throw_forward_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(_No_throw_sentinel_for<_Se, _It>);
+            _STL_INTERNAL_STATIC_ASSERT(default_initializable<iter_value_t<_It>>);
+
+            using _Ty = iter_value_t<_It>;
+            if constexpr (_Use_memset_value_construct_v<_It>) {
+                return _RANGES _Zero_range(_OFirst, _OLast);
+            } else if constexpr (is_nothrow_default_constructible_v<_Ty>) {
+                for (; _OFirst != _OLast; ++_OFirst) {
+                    ::new (static_cast<void*>(_Unfancy(_OFirst))) _Ty{};

```

Comment:
This needs _`voidify`_ magic as well, although we can get away with simply using:
```suggestion
                    _Construct_in_place(*_OFirst);
```
since value-initialization is ok (which is not the case for `uninitialized_default_meow`).

Reply:
I used voidify here so that it is more consistent